Create test cases for integrating index writer and reader

codermomo opened the pull request.

Oscar-0713 commented:
@codermomo pls update the submodule and check once again after fixing the gamma inconsistency issue
Then we can move on to the integration test. Thanks!

codermomo commented:
- Fixed bugs in plr index reader, writer, and iterator discovered during testing.
- Added new function `UpdateBinarySeekRange()` in `PLRBlockIter`.
- Added a new example of encoding and decoding.
- Added a test case for plr index iterator, using writer and iterator classes.
